kafka/README.md

```diff
@@ -0,0 +1,158 @@
+
+In this example, we demonstrate the implementation of end-to-end encryption for Kafka messages using encryption keys managed by Azure Managed Hardware Security Modules (mHSM). The key is only released when the Kafka consumer runs within a confidential container environment with azure attestation secret provisioning container injected into the pod.
+
+This example comprises four components: 
+
+Kafka Cluster: A simple kafka cluster deployed in the kafka namespace on an AKS cluster. 
+
+Kafka Producer: A kafka producer running as a vanilla k8s pod that sends encrypted user-configured messages using a public key to a kafka topic. 
+
+Kafka Consumer: A kafka consumer pod running with the kata-cc runtime class, equipped with a azure attestation secret provisioning container to retrieve the private key for decrypting encrypted kafka messages. 
+
+Web Service: Consumed messages are sent to a web service for display on a web UI. Messages, whether successfully decrypted or not, will be displayed. If not decrypted, they will appear as base64-encoded ciphertext.  
+
+### Step by Step Example 
+
+#### Enable Confidential Container on AKS cluster during creation.  
+
+```
+az aks create -g myResourceGroup -n myManagedCluster –kubernetes-version <1.24.0 and above> --os-sku AzureLinux –vm-size <VM sizes capable of nested SNP VM> --workload-runtime <kataCcIsolation> 
+```
+
+#### Enable workload identities on the cluster.  
+
+Update the AKS cluster using the az aks update command with the `--enable-oidc-issuer` parameter to use the OIDC Issuer.
+
+```
+export RESOURCE_GROUP="myResourceGroup" # This is the name of the resource group your AKS cluster resides 
+az aks update -g "${RESOURCE_GROUP}" -n myAKSCluster --enable-oidc-issuer --enable-workload-identity
+```
+
+Or append `--enable-oidc-issuer` `--enable-workload-identity` parameters to the end of your az aks create command so that the cluster is created to use the OIDC issuer. 
+
+#### Setup Federated Identity using Managed Identity as the Parent Resource 
+
+```
+export LOCATION="westcentralus" # This is the region of the resource group your AKS cluster resides 
+export SERVICE_ACCOUNT_NAMESPACE="default" # This is the kubernetes namespace you intend to run encfs workload
+export SERVICE_ACCOUNT_NAME="workload-identity-sa" 
+export SUBSCRIPTION="$(az account show --query id --output tsv)"
+export USER_ASSIGNED_IDENTITY_NAME="myIdentity" 
+export FEDERATED_IDENTITY_CREDENTIAL_NAME="myFedIdentity" 
+```
+
+Get the OIDC Issuer URL and save it to an environmental variable using the following command. 
+Replace the default value for the arguments -n, which is the name of the cluster.
+
+```
+export AKS_OIDC_ISSUER="$(az aks show -n aks-cluster-name -g "${RESOURCE_GROUP}" --query "oidcIssuerProfile.issuerUrl" -otsv)"
+```
+
+Create a managed identity in the same resource group your AKS cluster resides in and export the client_id of the managed identity
```

Comment:
I think "client_id" can be changed to "client ID" (or `USER_ASSIGNED_CLIENT_ID`)

Reply:
good catch